### PR TITLE
Add form

### PR DESCRIPTION
**Pull request by Zia123456**
### Project requirements
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+    @post.author = current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      flash[:notice] = 'Post created successfully!'
+      redirect_to user_posts_path(current_user)
+    else
+      flash[:alert] = "Couln't create post!"
+      render :new, status: :unprocessable_entity
+    end
```
Comment (Reem-lab) [inline app/controllers/posts_controller.rb:22]: - [x] Great job for adding the new and create but I can't see any forms in your views to let me be able to create a new post and I can't see the users even if I am putting the right route in the URL but I am not be able to see any posts it shows me this error could you solve it please? 💯🔥🤩✨

## I created some users and posts with `rails c` but I can't see ⬇⬇
image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @comment.author = current_user
+    @comment.post = Post.find(params[:post_id])
+    if @comment.save
+      flash[:notice] = 'Comment added successfully!'
+      redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
+    else
+      flash[:alert] = "Couln't add Comment!"
+      render :new, status: :unprocessable_entity
+    end
+  end
```
Comment (Reem-lab) [inline app/controllers/comments_controller.rb:17]: - [ ] Great job for adding the new and create but I can't see any forms in your views to let me be able to create a new comment and I can't see the users even if I am putting the right route in the URL but I am not be able to see any comments it shows me this error could you solve it please? 💯🔥🤩✨

## I created some users and posts and comments with `rails c` but I can't see ⬇⬇

image

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
-
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.order(created_at: :asc)
+  end
+
+  def new
+    @post = Post.new
+  end
```
Comment (Reem-lab) [inline app/controllers/posts_controller.rb:9]: - When I try to add a new post with this URL it shows me that can't create post! so kindly solve this error here ⬇✨🤩 and check why is prohibting from creating new posts 
image

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -17,4 +17,6 @@
 <button type="button" class="btn">
   <%= link_to "See All Posts", user_posts_path(@user) %>
 </button>
+
+<button type="button" class="btn"><%= link_to "Create a post", new_user_post_path(@user)  %></button>
```
Comment (Reem-lab) [inline app/views/users/show.html.erb:21]: - [x] I can't see any users displayed in your page even after creating users in rails console so I recommend to fix your views to let the user be able to see all of your view correctly 🤩✨🔥💯


image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @Zia123456, 👋🏻👋🏻 

Good job so far! 💯🔥✨🤩
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No Linters error 🤩
- followed Gitflow 🤩✨
- Great working on the functionality 🔥
- Professional README 💯💯


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Zia123456) [reply app/controllers/comments_controller.rb:17]: this is the correct path http://127.0.0.1:3000/users/1/posts/4
correct_path

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Zia123456) [reply app/controllers/posts_controller.rb:22]: there is not any user in your database with id = 1. I do not get any errors all features are working.
users

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (Zia123456) [reply app/controllers/posts_controller.rb:9]: I checked it is working, please first create some users, then check other features.
post

File: app/views/users/show.html.erb (hunk 4 above)
Comment (Zia123456) [reply app/views/users/show.html.erb:21]: users

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Zia123456) [reply app/controllers/posts_controller.rb:22]: please compare my design with the reference design, I just followed the reference design. 
users
single-user
post
single-post
reference

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Reem-lab) [reply app/controllers/comments_controller.rb:17]: Hi @Zia123456 I am sorry but I showed you what I saw on my local machine as well I sent messages to you to make sure
Could you send it again to review?

**Review by Reem-lab — APPROVED** [review]
# Hi @Zia123456,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯



## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- No Linters error 💯✨
- Great working on the functionality🔥🔥
- Professional README 🔥✨💯💯


## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**